staging/src/k8s.io/apiserver/pkg/registry/rest/create.go

```diff
@@ -235,3 +235,20 @@ func ValidateDeclaratively(ctx context.Context, obj runtime.Object, subresources
 		return field.ErrorList{field.InternalError(nil, fmt.Errorf("could not find requestInfo in context"))}
 	}
 }
+
+func ValidateUpdateDeclaratively(ctx context.Context, obj, oldObj runtime.Object, subresources ...string) field.ErrorList {
+	if requestInfo, found := genericapirequest.RequestInfoFrom(ctx); found {
+		groupVersion := schema.GroupVersion{Group: requestInfo.APIGroup, Version: requestInfo.APIVersion}
+		versionedObj, err := legacyscheme.Scheme.ConvertToVersion(obj, groupVersion)
+		if err != nil {
+			return field.ErrorList{field.InternalError(nil, fmt.Errorf("unexpected error converting to versioned type: %w", err))}
+		}
+		versionedOldObj, err := legacyscheme.Scheme.ConvertToVersion(oldObj, groupVersion)
+		if err != nil {
+			return field.ErrorList{field.InternalError(nil, fmt.Errorf("unexpected error converting to versioned type: %w", err))}
+		}
+		return legacyscheme.Scheme.ValidateUpdate(versionedObj, versionedOldObj, subresources...)
```

Comment:
Am I following this right?  This will call the registered funcs which take oldObj but don't do anything with it?  Is the assumption that things like +immutable will trigger them to use oldObj?

Reply:
That's right.  I've been plumbing in UpdateValidation as I go even though we don't have anything to register for it right now, so oldObj get's dropped in the generated code (until we improve that so we can support immutability and such)